### PR TITLE
Feature: Get user profile route

### PR DESCRIPTION
**Pull request by mariusjagminas**

### CODE REVIEW

File: src/users/users.controller.spec.ts [hunk 1]
```diff
@@ -428,4 +428,82 @@ describe('UsersController', () => {
...
+      email: mockedRequest.email,
+      password: mockedRequest.password,
+    });
+
+    const { accessToken } = service.getToken(id);
+
+    const result = await request(app.getHttpServer())
+      .get('/users/profile')
+      .set('Authorization', `Bearer ${accessToken}`)
+      .expect(404);
```
Comment (kowalski) [inline src/users/users.controller.spec.ts:501]: imo this is not a correct behaviour

We do have the user in db, just didn't create a record in userProfile for him.

Frontend is not to be blamed for this.
We should return 200 with empty profile (default values).

It's fine to create a placeholder record for this user as well

File: src/users/users.controller.spec.ts [hunk 2]
```diff
@@ -428,4 +428,82 @@ describe('UsersController', () => {
 
     expect(result.body).toEqual({ statusCode: 401, message: MSG.UNAUTHORIZED });
   });
+
+  it('should return user profile ', async () => {
+    const { id } = await factory.create<User>('User', {
```
Comment (kowalski) [inline src/users/users.controller.spec.ts:433]: You don't need to call factory for User and than for UserProfile.
If factory is done right you can just call it for UserProfile and the User record will be created for you by the power of magic

**Review by kowalski — CHANGES_REQUESTED** [review]

File: src/users/users.controller.spec.ts (hunk 1 above)
Comment (mariusjagminas) [reply src/users/users.controller.spec.ts:501]: Ok. An empty user profile placeholder will be created if there is no record in DB.

File: src/users/users.controller.spec.ts (hunk 2 above)
Comment (mariusjagminas) [reply src/users/users.controller.spec.ts:433]: Nice. I'll check this magic.

**Review by kowalski — APPROVED** [review]

**Pull request merged**